File: app/src/main/java/tv/spacedentist/android/view/SDButtonView.java
package tv.spacedentist.android.view;

import android.content.Context;
import android.graphics.Typeface;
import android.util.AttributeSet;
import android.util.Log;

import androidx.appcompat.widget.AppCompatButton;

/**
 * A custom Button so we can set the font
 */
public class SDButtonView extends AppCompatButton {
    private static final String TAG = SDButtonView.class.getSimpleName();

    public SDButtonView(Context context, AttributeSet attrs) {
        super(context, attrs);
        try {
            Typeface tf = Typeface.createFromAsset(context.getAssets(), "fonts/BebasNeue.ttf");
            setTypeface(tf);
        } catch (Exception e) {
            Log.e(TAG, "Couldn't load font", e);
        }
    }
}

File: app/src/main/java/tv/spacedentist/android/view/SDButtonView.kt
package tv.spacedentist.android.view

import android.content.Context
import android.graphics.Typeface
import android.util.AttributeSet
import android.util.Log

import androidx.appcompat.widget.AppCompatButton

/**
 * A custom Button so we can set the font
 */
class SDButtonView(context: Context, attrs: AttributeSet) : AppCompatButton(context, attrs) {

    init {
        try {
            val tf = Typeface.createFromAsset(context.assets, "fonts/BebasNeue.ttf")
            typeface = tf
        } catch (e: Exception) {
            Log.e(TAG, "Couldn't load font", e)
        }

    }

    companion object {
        private val TAG = "SDButtonView"
    }
}
